src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
+}
+
+/// FileSystem to load files relative to the root
+pub struct LocalFileRepository {
+    root: PathBuf,
+}
+
+impl FileRepository for LocalFileRepository {
+    fn read_template_file(&self, relative_path: &TemplateLocation) -> Result<String> {
+        let path = self.root.clone().join(relative_path);
+
+        if !path.exists() {
+            return Err(Error::from(&*format!("{:?} does not exist", path)));
+        }
+        let mut file = try!(File::open(path));
+
+        let mut content = String::new();
+        file.read_to_string(&mut content)?;
+        Ok(content)
+    }
+}
+
 /// Options that `liquid::parse` takes
-#[derive(Default)]
 pub struct LiquidOptions {
     /// Holds all custom block-size tags
     pub blocks: HashMap<String, Box<Block>>,
     /// Holds all custom tags
     pub tags: HashMap<String, Box<Tag>>,
     /// The path to which paths in include tags should be relative to
-    pub file_system: Option<PathBuf>,
+    pub file_repository: Box<FileRepository>,
```

Comment:
Box or ref?

I think box is fine.  A small part of me wishes this was instead a ref but the lifetime would then have to be added through the entire code base and might get nasty.  Maybe something we can re-evaluate for later.